test/test_jit.py

```diff
@@ -1220,42 +1248,408 @@ def where(c, a, b):
+            if a > b:
+                a = a + b
+            return a
+
+        a, batch_a = self.rand_batch(4, ())
+        b, batch_b = self.rand_batch(4, ())
+        res_batch = batch_if(batch_a, batch_b)
+        res = [single_if(a[j], b[j]) for j in range(4)]
+        self.assertEqual(res, res_batch.examples())
+
+        @torch.jit.script
+        def batch_if(a, b):
+            if a > b:
+                a = a + b
+            return a
+
+        graph = torch.to_batch_graph(batch_if.graph)
+        self.assertExpected(str(graph))
+
+    def test_if_noelse_with_scalar(self):
+        @torch.jit.batch(batch_size=4)
+        def batch_if(a, b):
+            if a > 0.1:
+                a = a + b
+            return a
+
+        def single_if(a, b):
+            if a > 0.1:
+                a = a + b
+            return a
+
+        a, batch_a = self.rand_batch(4, ())
+        b, batch_b = self.rand_batch(4, ())
+        res_batch = batch_if(batch_a, batch_b)
+        res = [single_if(a[j], b[j]) for j in range(4)]
+        self.assertEqual(res, res_batch.examples())
+
+        @torch.jit.script
+        def batch_if(a, b):
+            if a > 0.1:
+                a = a + b
+            return a
+
+        graph = torch.to_batch_graph(batch_if.graph)
+        self.assertExpected(str(graph))
+
+    def test_while(self):
+        @torch.jit.batch(batch_size=4)
+        def batch_while(a, b):
+            while a > b:
+                a = a - b
+            return a
+
+        def single_while(a, b):
+            while a > b:
+                a = a - b
+            return a
+
+        a, batch_a = self.rand_batch(4, ())
+        b = [torch.abs(torch.rand(1)) for i in range(4)]
+        batch_b = BatchTensor(b, torch.tensor([]).byte())
+        res_batch = batch_while(batch_a, batch_b)
+        res = [single_while(a[j], b[j]) for j in range(4)]
+        self.assertEqual(res, res_batch.examples())
+
+        @torch.jit.script
+        def batch_while(a, b):
+            while a > b:
+                a = a - b
+            return a
+
+        graph = torch.to_batch_graph(batch_while.graph)
+        self.assertExpected(str(graph))
+
+    def test_for(self):
+        @torch.jit.batch(batch_size=4)
+        def batch_for(x, y):
+            for _ in range(10):
+                x = x + y
+            return x
+
+        def single_for(x, y):
+            for _ in range(10):
+                x = x + y
+            return x
+
+        a, batch_a = self.rand_batch(4, ())
+        b, batch_b = self.rand_batch(4, ())
+        res_batch = batch_for(batch_a, batch_b)
+        res = [single_for(a[j], b[j]) for j in range(4)]
+        self.assertEqual(res, res_batch.examples())
+
+        @torch.jit.script
+        def batch_for(x, y):
+            for _ in range(10):
+                x = x + y
+            return x
+
+        graph = torch.to_batch_graph(batch_for.graph)
+        self.assertExpected(str(graph))
+
+    def test_lstm(self):
```

Comment:
For these tests, instead of repeating the implementation twice, you can simply use the decorator as a function:

```
def foo(x):
  return x

foo_batched = torch.jit.batch(batch_size=4)(foo)